### PR TITLE
Various diagnostic UI improvements

### PR DESCRIPTION
**Pull request by erri120**
Part of #1091.

- Fixes the horizontal line coloring:

- Adds chips with counts and filters
- Adds text for no diagnostics
- Adds the summary as a header to the details page:
- Adds some basic Markdown styling:

### CODE REVIEW

**Comment by Al12rs** [comment]
Could you add a "No Entries" text if there are no diagnostics to see in a given filtered view? 
Use a Subdued foreground color and pick a text style

**Comment by Al12rs** [comment]
Did you figure out the chip behavior with Design?

**Comment by erri120** [comment]
> Did you figure out the chip behavior with Design?

Yeah, they now work similar to tabs, meaning you can only have one filter active at a time.

File: src/NexusMods.App.UI/Pages/Diagnostics/List/IDiagnosticListViewModel.cs [hunk 1]
```diff
@@ -10,4 +13,23 @@ public interface IDiagnosticListViewModel : IPageViewModelInterface
...
+    ReactiveCommand<Unit, Unit> ShowAllCommand { get; }
+}
+
+[Flags]
+public enum DiagnosticFilter
+{
+    None = 0,
+    Critical = 1,
+    Warnings = 2 << 0,
+    Suggestions = 2 << 1,
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Pages/Diagnostics/List/IDiagnosticListViewModel.cs:34]: For consistency, use `1 << x` in all non 0 values. Should be clearer

**Review by Al12rs — APPROVED** [review]

**Pull request merged**